src/main/scala/com/metl/model/BackendAdaptor.scala

```diff
@@ -550,3 +551,28 @@ class ResourceCachingAdaptor(sc:ServerConfiguration,cacheConfig:CacheConfig) ext
     super.isReady
   }
 }
+
+class D2LCachingAdaptor(cacheConfig:CacheConfig) {
```

Comment:
5s first time, 2s second time.  I don't think very much of that is computation.  I don't know what our locking strategy is - I'm only doing sum(count()) in SQL so I would hope that doesn't lock rows, and if it does then only for a miniscule time.